src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala

```diff
@@ -181,8 +200,14 @@ object ConductRSandbox extends AutoPlugin {
 
     val debugPorts = state.value.get(WithDebugAttrKey).fold(Set.empty[Int])(_ => debugPort.?.all(filter).value.flatten.toSet)
 
+    val roles = (conductrRoles in Global).value match {
+      case Nil           => Seq(BundleKeys.roles.?.map(_.getOrElse(Set.empty)).all(filter).value.reduce(_ ++ _))
```

Comment:
No, it never can be empty. Tested it with no roles set. As a result we get back a List which includes empty sets:

```scala
List(Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set())
```

Same for endpoints. By default `web` is set for endpoints`. If the user overwrites it to an `Map.empty` we get this back:

```scala
List(Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map(), Map())
roles: List(Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set(), Set())
```